### PR TITLE
Fix ManagedHandler env var / AppContext handling

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/26466
cc: @eerhardt, @wfurt, @geoffkizer

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs [hunk 1]
```diff
@@ -12,40 +12,45 @@ namespace System.Net.Http
...
 
-                if (AppContext.TryGetSwitch(AppCtxManagedHandlerSettingName, out bool isManagedEnabled) && isManagedEnabled)
+                // AppContext switch wasn't used. Check the environment variable to see if it's been set to true.
+                string envVar = Environment.GetEnvironmentVariable(ManagedHandlerEnvironmentVariableSettingName);
+                if (envVar != null && (envVar.Equals("true", StringComparison.OrdinalIgnoreCase) || envVar.Equals("1")))
                 {
                     return true;
                 }
 
+                // TODO #23166: Remove the following TLS check assuming the type is exposed publicly.  If it's not,
```
Comment (eerhardt) [inline src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs]: Dumb question:  Why not just delete it now?  Is there a reason for keeping it?  Who is using it?

Comment (stephentoub) [reply src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs]: We are.  We need a non-process-wide mechanism to enable the handler, e.g. we use it in ClientWebSocket, and when ClientWebSocket creates the HttpClientHandler it must use ManagedHandler.

Comment (eerhardt) [reply src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs]: Don't those other places need to be updated then?

https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=COMPlus_UseManagedHttpClientHandler&type=

Comment (stephentoub) [reply src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs]: Yes, yes they do.

**Comment by stephentoub** [comment]
@dotnet-bot test Linux x64 Release Build please (https://github.com/dotnet/corefx/issues/25676)

**Comment by wfurt** [comment]
looks good. thanks @stephentoub

**Pull request merged**